README.md

```diff
@@ -16,9 +16,28 @@ This is a monorepo that can hold all of the Synthetic Canary applications, each
+* Creating a new canary is as simple as creating a root level folder
+* A single canary may perform checking for one or many applications; it provides a method but is agnostic to what application it's used to check
+* Each canary in this repository (that is, each top-level folder) will have a few hard requirements:
+  * A `.runtime` file in the folder that designates the expected Synthetics Canary runtime (see [AWS: Synthetics runtime versions](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Synthetics_Canaries_Library.html))
+  * A `Makefile` with (at a minimum) `install`, `update`, and `test` targets (as these are expected by the root [Makefile](./Makefile))
+  * A `zipmanifest.txt` file that designates how the ZIP will be built for loading into a Synthetics Canary (different Synthetic Canary runtimes have different requirements for the structure of the ZIP file)
+
+## Deployment Overview
+
+Deploying (or building) in this repository simply means copying a ZIP file of an application to a shared S3 bucket in AWS. That ZIP file is like a blueprint (or template) application that can be loaded into a Synthetic Canary that is built by Terraform to monitor an HTTP/HTTPS endpoint. It will be helpful to have both a `zip` and a `dev-publish` targets in the canary folder `Makefile` so that it will be easy for the developer to deploy the ZIP to Dev1 for testing in AWS.
+
+The rest of the deployment process is handled by GitHub Actions, following the MITL typical GitHub-flow:
+
+* Opening a PR to `main` will deploy the ZIP to our dev AWS account (and create/update SSM Parameters)
+* Merging a PR to `main` will deploy the ZIP to our stage AWS account (and create/update SSM Parameters)
+* Tagging a release on `main` will deploy the ZIP to our prod AWS account (and create/update SSM Parameters)
+
+## General Overview
+
+The general idea of this repository is to de-couple the actual CloudWatch Synthetic Canary application code from the Terraform repositories that will deploy the Canary to monitor web services. Since the application code is decoupled from the Terraform repositories, it means that we can write fewer applications that are more generic (they are like blueprints or templates). We can customize the generic applications in each Terraform repository that builds the Canary infrastructure by using environment variables that can be set by Terraform. This requires that we export some critical information from the deployment process for this repository to SSM Parameter Store so that the dependent Terraform repositories can read this information to build the infrastructure for the Canary correctly.
```

Comment:
:chef's kiss:

Thanks for adding this!  I feel like it's filling the gap I had on my first pass.